python/plugins/processing/algs/qgis/RandomPointsOnLines.py

```diff
@@ -0,0 +1,257 @@
+                                                       QgsProcessingParameterNumber.Integer,
+                                                       None, True, 1, 1000000000))
+        self.addParameter(QgsProcessingParameterFeatureSink(self.OUTPUT,
+                                                            self.tr('Random points'),
+                                                            type=QgsProcessing.TypeVectorPoint))
+        self.addOutput(QgsProcessingOutputNumber(self.OUTPUT_POINTS,
+                                                 self.tr('Number of point generated')
+                                                 ))
+
+    def processAlgorithm(self, parameters, context, feedback):
+        source = self.parameterAsSource(parameters, self.INPUT, context)
+        if source is None:
+            raise QgsProcessingException(self.invalidSourceError(parameters,
+                                                                 self.INPUT))
+
+        pointCount = self.parameterAsInt(parameters, self.POINTS_NUMBER,
+                                         context)
+        minDistance = self.parameterAsDouble(parameters, self.MIN_DISTANCE,
+                                             context)
+        randSeed = self.parameterAsInt(parameters, self.RANDOM_SEED, context)
+        MaxTriesPerPoint = self.parameterAsInt(parameters,
+                                               self.MAXTRIESPERPOINT,
+                                               context)
+        fields = QgsFields()
+        fields.append(QgsField('id', QVariant.Int, '', 10, 0))
+
+        (sink, dest_id) = self.parameterAsSink(parameters, self.OUTPUT,
+                                               context, fields,
+                                               QgsWkbTypes.Point,
+                                               source.sourceCrs(),
+                                               QgsFeatureSink.RegeneratePrimaryKey)
+        if sink is None:
+            raise QgsProcessingException(
+                self.invalidSinkError(parameters, self.OUTPUT))
+
+        totNPoints = 0  # The total number of points generated
+        missedPoints = 0  # The number of misses (too close)
+        featureCount = source.featureCount()
+        total = 100.0 / (pointCount * featureCount) if pointCount else 1
+        if randSeed:
+            random.seed(randSeed)
+        else:
+            random.seed()
+
+        index = QgsSpatialIndex()
+        points = dict()
+
+        maxIterations = pointCount * MaxTriesPerPoint
+        # Go through all the features of the layer
+        for f in source.getFeatures():
+            if feedback.isCanceled():
+                break
+            lineGeoms = []  # vector of line elements (>=1 for multi)
+            fGeom = f.geometry()  # get the geometry
+            if fGeom is None:
+                feedback.reportError('Null geometry - skipping!', False)
```

Comment:
We usually just skip these without warning

Reply:
I think the user has the right to know. Could perhaps be more silent...

Reply:
Changed to pushInfo.